python/ray/train/lightning/lightning_trainer.py

```diff
@@ -382,13 +383,13 @@ def __init__(
         preprocessor: Optional[Preprocessor] = None,
         resume_from_checkpoint: Optional[Checkpoint] = None,
     ):
-        run_config = run_config or RunConfig()
+        run_config = deepcopy(run_config) or RunConfig()
```

Comment:
I think we can just make a shallow copy, since we're just pointing the ckpt config to something else.

Reply:
Done